lib/core/ingestion/ingester.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'utility/logger'
+
+module Core
+  module Ingestion
+    class Ingester
+      def initialize(sink_strategy)
+        @sink_strategy = sink_strategy
+        @ingested_count = 0
+        @ingested_volume = 0
+        @deleted_count = 0
+      end
+
+      def ingest(document)
+        unless document&.any?
+          Utility::Logger.warn('Connector attempted to ingest an empty document, skipping')
+          return
+        end
+
+        serialized_document = @sink_strategy.serialize(document)
```

Comment:
I think we should add a base class for `ConsoleSink` and `EsSink` to make sure `serialize` method is implemented, as well as `ingest` and `delete`

Reply:
At this point I feel like dropping ConsoleSink and BaseSink and just leaving the class that does all ingestion operations. ConsoleSink is not used and makes refactoring harder.